lib/src/working_copy.rs

```diff
@@ -911,8 +919,33 @@ impl TreeState {
                     file_state.mark_executable(executable);
                     stats.updated_files += 1;
                 }
+                Diff::Modified(TreeValue::GitSubmodule(_old_id), after) => {
```

Comment:
matching on `before` here makes the code less cluttered than adding more `if` clauses in the other branch

Reply:
I think we should still try to write the `after` state to disk if we can successfully remove the file (because the user had not put a directory there) or if the path doesn't exist at all (because we don't write anything for submodules).